TechnicoBackend/Controllers/AuthController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TechnicoBackend.Repositories;
+using TechnicoBackend.Models;
+using TechnicoBackend.DTOs;
+
+namespace TechnicoBackend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class AuthController : ControllerBase
+    {
+        private readonly UserRepository _userRepository;
+
+        public AuthController(UserRepository userRepository)
+        {
+            _userRepository = userRepository;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserDTO userDto)
+        {
+            try
+            {
+                var users = await _userRepository.GetAllAsync();
+                if (users.Any(u => u.Email == userDto.Email))
+                {
+                    return BadRequest("Το email χρησιμοποιείται ήδη.");
+                }
+
+                var user = new User
```

Comment:
Consider passing the userDto to a Service. Consider using Controller to only receive the data and call the service.